### PR TITLE
ROX-12016, ROX-13267, ROX-12806, ROX-13117: Automate preparations for manual upgrade tests and various fixes

### PR DESCRIPTION
**Pull request by tommartensen**
## Description

This PR adds automation to prepare clusters for the manual upgrade tests (upstream release process). 
It also introduces various fixes to finish-release, release notes creation, JIRA issue checking and the local test bed.

Specifically we transfer the following PRs from our development repo:

* https://github.com/stackrox/test-gh-actions/pull/65
* https://github.com/stackrox/test-gh-actions/pull/66
* https://github.com/stackrox/test-gh-actions/pull/67
* https://github.com/stackrox/test-gh-actions/pull/69
* https://github.com/stackrox/test-gh-actions/pull/70
* https://github.com/stackrox/test-gh-actions/pull/71

## Checklist
- [ ] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Testing performed in development repository, see linked Pull Requests.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Review by vikin91 — APPROVED** [review]
Is there a chance that we get at least a smoke-test in this repo after merging so many small parts?

**Comment by tommartensen** [comment]
> Is there a chance that we get at least a smoke-test in this repo after merging so many small parts?

Is 3.72.2 enough smoke test?

**Comment by vikin91** [comment]
> > Is there a chance that we get at least a smoke-test in this repo after merging so many small parts?
> 
> Is 3.72.2 enough smoke test?

Yes 😄

File: .github/workflows/cut-rc.yml
Comment (janisz) [inline .github/workflows/cut-rc.yml:270]: `needs.variables.outputs.milestone` ?

Comment (vikin91) [reply .github/workflows/cut-rc.yml:270]: @tommartensen would you be able to assist Tomek here?

Comment (tommartensen) [reply .github/workflows/cut-rc.yml:270]: Yes, going to cherry-pick the fix for this from https://github.com/stackrox/stackrox/pull/3726.

Comment (tommartensen) [reply .github/workflows/cut-rc.yml:270]: Please review https://github.com/stackrox/stackrox/pull/3877. If you don't want to wait for the merge, you may run the "Cut RC" workflow manually and choose this branch for the workflow definition from the dropdown.

**Pull request merged**
